x-pack/plugins/index_management/public/application/lib/data_streams.tsx

```diff
@@ -38,3 +39,17 @@ export const isSelectedDataStreamHidden = (
       ?.hidden
   );
 };
+
+export const getLifecycleValue = (lifecycle: DataStream['lifecycle']) => {
+  if (!lifecycle?.enabled) {
```

Comment:
Will lifecycle always have the `enabled` flag? In other words, can `lifecycle.enabled` be undefined, and, if so, is the default behavior that it is disabled? Will this change once DSL goes GA?

Reply:
With this line we address two scenarios: if lifecycle is not defined then its `disabled` and if its defined, but its not enabled its also `disabled`.


If a DS has lifecycle, it will have an `enabled` attribute defined, I believe the default behaviour would be that if you set the lifecycle to empty:
```
PUT _data_stream/b-data-stream/_lifecycle 
{

}
```

It will become `lifecycle.enabled: true` which means that it will have infinite data retention.